src/flanders/core.cljc

```diff
@@ -62,6 +62,21 @@
    (merge opts
           {:type t})))
 
+(defn either
+  {:arglists '([choices* & {:as [opts]}])}
+  [& args]
+  (let [[choices options] (split-with (complement keyword?) args)]
+    (assert (seq choices) "either expects at least one choice")
+    ;; else
+    (ft/map->EitherType
+     (into {:choices choices}
+           (keep
+            (fn [[k v]]
+              (if (keyword? k)
+                (if (not (= k :choices))
+                  [k v])))
+            (partition-all 2 options))))))
```

Comment:
I think this proposal is superior because I tend to find that `[& {:as opts}]` style arguments are not as useful as passing a map and they're error prone. If you'll notice, the code I submitted manually implements the `[& {:as opts}]` but side steps the potential irritation of a `No value supplied for key: :foo` error via the `partition-all`. 😄 

Though I dislike `[& {:as opts}]` style arguments, the rest of the namespace is designed this way. Should we break with the established convention in this case?

Reply:
I made a mistake in my proposal:

```clojure
(defn either
  {:arglists '([choices* & {:as [opts]}])}
  [& {:keys [choices] :as args}]
  (assert (seq choices) "either expects at least one choice")
  (ft/map->EitherType args))
```

The `&` character was missing

Even if I dislike the `[& {:as opts}]` style arguments I don't want to break the current convention in this namespace.